framework/src/postprocessors/SideDiffusiveFluxIntegral.C

```diff
@@ -41,31 +47,44 @@ template <bool is_ad, typename T>
 SideDiffusiveFluxIntegralTempl<is_ad, T>::SideDiffusiveFluxIntegralTempl(
     const InputParameters & parameters)
   : SideIntegralVariablePostprocessor(parameters),
-    _diffusivity(getParam<MaterialPropertyName>("diffusivity")),
-    _diffusion_coef(getGenericMaterialProperty<T, is_ad>(_diffusivity))
+    _diffusion_coef(isParamValid("diffusivity")
+                        ? &getGenericMaterialProperty<T, is_ad>("diffusivity")
+                        : nullptr),
+    _functor_diffusion_coef(isParamValid("functor_diffusivity")
+                                ? &getFunctor<Moose::GenericType<T, is_ad>>("functor_diffusivity")
+                                : nullptr)
 {
+  if (_fv && !isParamValid("functor_diffusivity"))
+    mooseError(
+        "For a finite volume variable, the parameter 'functor_diffusivity' must be provided");
+  if (!_fv && !isParamValid("diffusivity"))
+    mooseError("For a finite element variable, the parameter 'diffusivity' must be provided");
 }
 
 template <bool is_ad, typename T>
 Real
-SideDiffusiveFluxIntegralTempl<is_ad, T>::computeQpIntegral()
+SideDiffusiveFluxIntegralTempl<is_ad, T>::computeFaceInfoIntegral(const FaceInfo * const fi)
 {
-  if (_fv)
-  {
-    // Get the face info
-    const FaceInfo * const fi = _mesh.faceInfo(_current_elem, _current_side);
-    mooseAssert(fi, "We should have a face info");
+  // Get the gradient of the variable on the face
+  const auto grad_u = MetaPhysicL::raw_value(_fv_variable->gradient(
+      Moose::FV::makeCDFace(*fi, Moose::FV::faceArgSubdomains(*_fv_variable, *fi))));
+  ;
 
-    // Get the gradient of the variable on the face
-    const auto & grad_u = MetaPhysicL::raw_value(_fv_variable->adGradSln(*fi));
+  // FIXME Get the diffusion coefficient on the face, see #16809
+  return -diffusivityGradientProduct(
+             grad_u,
+             MetaPhysicL::raw_value((*_functor_diffusion_coef)(Moose::FV::makeCDFace(
+                 *fi, Moose::FV::faceArgSubdomains(*_functor_diffusion_coef, *fi))))) *
+         (_fv_variable->hasBlocks(fi->elemSubdomainID()) ? fi->normal() : Point(-fi->normal()));
```

Comment:
so this means we set the normal of the flux based on the domain of definition of the variable, not based on the sideset definition.
I m ok with that, seems like the user would be more likely to get variable definitions right than sideset orientation, but we should add a note in the doc file

Reply:
Thinking about this, I really don't like the way I had this. For an internal face, one could imagine an element numbering scheme in which for one face, the lower element subdomain ID is on the "left" and then for the next face, the lower element subdomain ID is on the "right". In such a case your normal direction would change between the two faces. I think it's better to just go with the `_normals` member